hack/filter-crd/main.go

```diff
@@ -152,46 +117,12 @@ func checkSliceChain(s []interface{}, chain []string) []interface{} {
 	return s
 }
 
-func removeFromSlice(s []interface{}, d map[interface{}]interface{}) []interface{} {
-	newSlice := []interface{}{}
-
-	for _, sliceVal := range s {
-		if !reflect.DeepEqual(sliceVal, d) {
-			newSlice = append(newSlice, sliceVal)
-		}
-	}
-
-	s = newSlice
-	return s
-}
-
 func loadVariant() {
 	variant := ""
 	flag.StringVar(&variant, "variant", "", "variant of remove rules")
 	flag.Parse()
 
-	if variant == "cert-manager-legacy" {
-		// These are the keys that the script will remove for OpenShift 3 and older Kubernetes compatibility
-		removeKeys = []string{
-			"spec/preserveUnknownFields",
-			"spec/validation/openAPIV3Schema/type",
-			"spec/versions/[]/schema/openAPIV3Schema/type",
-			"spec/conversion",
-			// This field exists on the Issuer and ClusterIssuer CRD
-			"spec/validation/openAPIV3Schema/properties/spec/properties/acme/properties/solvers/items/properties/dns01/properties/webhook/properties/config/x-kubernetes-preserve-unknown-fields",
-			"spec/versions/[]/schema/openAPIV3Schema/properties/spec/properties/acme/properties/solvers/items/properties/dns01/properties/webhook/properties/config/x-kubernetes-preserve-unknown-fields",
-			// This field exists on the Challenge CRD
-			"spec/validation/openAPIV3Schema/properties/spec/properties/solver/properties/dns01/properties/webhook/properties/config/x-kubernetes-preserve-unknown-fields",
-			"spec/versions/[]/schema/openAPIV3Schema/properties/spec/properties/solver/properties/dns01/properties/webhook/properties/config/x-kubernetes-preserve-unknown-fields",
-		}
-
-		// only retain the `v1` version in the CRD
-		retainElementForValue = map[string]string{
-			"spec/versions/[]/name": "v1",
-		}
-
-		singleCRDVersion = true
-	} else if variant == "no-helm" {
+	if variant == "no-helm" {
```

Comment:
This all looks a bit weird now that there is only one "variant" to handle.

Perhaps a comment or two at the top of this file explaining what this script now does and what other variants might be added in the future.
And an example of how the command should be run.

Reply:
Added a comment with the history